Nure.UnitTests/PullRequest/Bitbucket/BitbucketPullRequestWriterTests.cs

```diff
@@ -0,0 +1,80 @@
+using System.Collections.Generic;
+using Moq;
+using Nure.Configuration;
+using Nure.PullRequest.Bitbucket;
+using Nure.PullRequest.Bitbucket.Models;
+using Xunit;
+
+namespace Nure.UnitTests.PullRequest.Bitbucket
+{
+    public class BitbucketPullRequestWriterTests
+    {
+        private const string A_BRANCH_NAME = "almost-heaven/west-verginia";
+        private const string A_CURRENT_USER_ID = "a";
+        private const string ANOTHER_USER_ID = "b";
+
+        private readonly Mock<IBitbucketClient> m_ClientMock = new Mock<IBitbucketClient>();
+
+        [Fact]
+        public void ValidPullRequestIsSent()
+        {
+            var bitbucketPullRequest = new BitbucketPullRequest();
+            var reviewers = GivenDefaultReviewers();
+            var nureOptions = GivenNureOptions();
+
+            m_ClientMock.Setup(client => client.GetCurrentUser()).Returns(new BitbucketUser());
+            m_ClientMock.Setup(client => client.GetDefaultReviewers()).Returns(reviewers);
+            m_ClientMock.Setup(client => client.SendPullRequest(It.IsAny<BitbucketPullRequest>()))
+                .Callback<BitbucketPullRequest>(pr => bitbucketPullRequest = pr);
+
+            var pullRequestWriter = new BitbucketPullRequestWriter(nureOptions, m_ClientMock.Object);
+            pullRequestWriter.WritePullRequest(A_BRANCH_NAME);
+
+            m_ClientMock.Verify(client => client.GetCurrentUser());
+            m_ClientMock.Verify(client => client.GetDefaultReviewers());
+            m_ClientMock.Verify(client => client.SendPullRequest(It.IsAny<BitbucketPullRequest>()));
+            
+            Assert.Equal(nureOptions.PullRequestTitle, bitbucketPullRequest.Title);
+            Assert.Equal(nureOptions.PullRequestDescription, bitbucketPullRequest.Description);
+            Assert.Equal(A_BRANCH_NAME, bitbucketPullRequest.SourceBranch.Branch.Name);
+            Assert.Equal(nureOptions.DefaultBranch, bitbucketPullRequest.DestinationBranch.Branch.Name);
+            Assert.Equal(reviewers, bitbucketPullRequest.Reviewers);
+        }
+
+        [Fact]
+        public void CurrentUserIsFilteredFromDefaultReviewers()
```

Comment:
Good catch. Were you being added as a reviewer because you were using your creds?

Reply:
Not quite -- it looks like bitbucket's API does not allow the creation of PRs where you're including yourself as a reviewer. I was getting access denieds.